dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/DataOkHttpUploaderV2.kt

```diff
@@ -34,17 +34,18 @@ internal abstract class DataOkHttpUploaderV2(
 
     private val uploaderName = javaClass.simpleName
 
+    internal val clientToken = if (isValidHeaderValue(rawClientToken)) rawClientToken else ""
+    internal val source: String = sanitizeHeaderValue(rawSource)
+    internal val sdkVersion: String = sanitizeHeaderValue(rawSdkVersion)
+
     private val userAgent by lazy {
-        System.getProperty(SYSTEM_UA).let {
-            if (it.isNullOrBlank()) {
-                "Datadog/$sdkVersion " +
+        sanitizeHeaderValue(System.getProperty(SYSTEM_UA))
+            .ifBlank {
+                "Datadog/${sanitizeHeaderValue(sdkVersion)} " +
```

Comment:
not even sure if the `sdkVersion` should be sanitized at all

Reply:
The `sdkVersion` will possibly be updatable from crossplatform SDKs, meaning it will be using public APIs. So in theory, it could have an invalid character yes